presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/ExpressionMatcher.java

```diff
@@ -101,7 +102,7 @@ public Optional<VariableReferenceExpression> getAssignedVariable(PlanNode node,
             }
         }
 
-        List<Object> matches = matchesBuilder.build();
+        Set<Object> matches = new HashSet<>(matchesBuilder.build());
```

Comment:
can you explain the reason for this change?

Reply:
In my optimization, a new projection node will be added to fan out the output of the identical aggregation function, i.e. from `sum_1 := sum(col) sum_2 := sum(col)` to `sum_1 := sum(col)` and `sum_1 := sum_1 sum_2 := sum1`. matchesBuilder will collect the two `sum_1` at the right side of the assignments, and fail the following assertions "matches.size() < 2". I think fan out assignments is a legit plan and should not be failed, hence add the change here.

Reply:
Got it. Yeah, I agree it's a valid plan. But this change essential makes the >2 check irrelevant anyway.

can we instead remove the check and have the matcher be able to match a list or set of variables or something?

Reply:
No, it does not make this check irrelevant. If the variable references collected in the list are not the same, the cardinality will be >= 2 after converting to a set (although I am not sure how one expression gets mapped to multiple variableReferences, but looks like that this is something we need to avoid and throw an error for).